jib-core/src/test/java/com/google/cloud/tools/jib/blob/BlobTest.java

```diff
@@ -66,6 +68,35 @@ public void testFromBlobWriter() throws IOException {
     verifyBlobWriteTo(expected, Blobs.from(writer));
   }
 
+  @Test
+  public void testWriteToFileWithLock_newFile() throws IOException {
+    String expected = "crepecake";
+    File file = File.createTempFile("blob", "bin");
```

Comment:
For consistency in this codebase, let's prefer `java.nio` for files over `java.io` - so `Files.createTempFile` and `Files.delete`